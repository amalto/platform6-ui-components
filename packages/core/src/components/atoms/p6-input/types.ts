export enum P6InputType {
  email = 0,
  number = 1,
  password = 2,
  search = 3,
  tel = 4,
  text = 5,
  url = 6,
}
