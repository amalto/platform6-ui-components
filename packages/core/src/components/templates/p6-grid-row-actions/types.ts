import { IconName, IconPrefix } from '@fortawesome/fontawesome-svg-core';
import { EventEmitter } from '@stencil/core';
import { Mode } from '../../../shared/types';
import { RowId } from '../../organisms/p6-grid/core/entities';

export interface ActionsOptions {
  create: { disabled: boolean; canUse: boolean };
  view: { disabled: boolean; canUse: boolean };
  edit: { disabled: boolean; canUse: boolean };
  rename: { disabled: boolean; canUse: boolean };
  delete: { disabled: boolean; canUse: boolean };
  export: { disabled: boolean; canUse: boolean };
  [action: string]: { disabled: boolean; canUse: boolean };
}

export interface Action {
  id: keyof ActionsOptions;
  title: string;
  icon: IconName;
  iconPrefix?: IconPrefix;
  mode?: Mode;
  emitter: () => EventEmitter<{ rowId: RowId }> | undefined;
}
