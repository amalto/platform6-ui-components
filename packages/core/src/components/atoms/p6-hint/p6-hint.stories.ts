import { Components } from '../../../components';
import { ComponentProps, getElement, makeModeStory, makeStory, Props } from '../../../shared/storybook';
import { Mode } from '../../../shared/types';

const component = 'p6-hint';

export default {
  title: 'Atoms/Hint',
  component,
};

const componentProps: ComponentProps = ['mode'];

const getStoryField = (text: string, props?: Props<Components.P6Hint>): HTMLElement => getElement(component, text, props);

// Default
export const Default = makeStory<{ label: string; mode: Mode }>({
  componentProps,
  args: { label: 'hint', mode: Mode.success },
  builder: ({ label, ...props }): HTMLElement => getStoryField(label, { ...props }),
});

// --- Mode
export const Modes = makeModeStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField(key, {
      mode: value,
    }),
});
