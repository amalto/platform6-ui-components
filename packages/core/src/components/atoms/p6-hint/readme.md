# p6-hint

<!-- Auto Generated Below -->

## Properties

| Property | Attribute | Description              | Type                                                                                       | Default        |
| -------- | --------- | ------------------------ | ------------------------------------------------------------------------------------------ | -------------- |
| `mode`   | `mode`    | set the mode of the hint | `Mode.danger \| Mode.default \| Mode.info \| Mode.primary \| Mode.success \| Mode.warning` | `Mode.default` |

## Dependencies

### Used by

- [p6-field](../../molecules/p6-field)

### Graph

```mermaid
graph TD;
  p6-field --> p6-hint
  style p6-hint fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
