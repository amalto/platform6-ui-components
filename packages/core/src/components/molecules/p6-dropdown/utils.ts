export function addDropdownClass(child: Element): Element {
  if (child.nodeName === 'HR') {
    child.classList.add('dropdown-divider');
    return child;
  }

  child.classList.add('dropdown-item');
  return child;
}
