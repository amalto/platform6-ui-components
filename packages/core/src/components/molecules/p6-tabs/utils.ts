export function getTabId(id: string): string {
  return `${id}-tab`;
}

export function isTabValid(tab: Element): tab is HTMLElement {
  return 'title' in tab && 'id' in tab;
}
