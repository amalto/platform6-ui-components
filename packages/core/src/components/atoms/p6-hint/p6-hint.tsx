import { Component, ComponentInterface, h, Host, JSX, Prop } from '@stencil/core';
import { Mode } from '../../../shared/types';
import { getModeClass } from '../../../utils/classes';

@Component({
  tag: 'p6-hint',
  styleUrl: 'p6-hint.scss',
  shadow: true,
})
export class P6Hint implements ComponentInterface {
  /**
   * set the mode of the hint
   */
  @Prop() mode: Mode = Mode.default;

  render(): JSX.Element {
    return (
      <Host
        class={{
          ...getModeClass(this.mode),
        }}
      >
        <slot />
      </Host>
    );
  }
}
