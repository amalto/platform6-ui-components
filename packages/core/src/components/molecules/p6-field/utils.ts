import { isP6Control, isP6NativeControl, P6Control, P6NativeControl } from '../../../shared/form/control';
import { isInDefaultSlot } from '../../../utils/component';

// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
export function isControl(elmt: unknown): elmt is P6Control<unknown> | P6NativeControl {
  return isP6Control(elmt) || isP6NativeControl(elmt);
}

export function getControl(children: HTMLCollection): P6Control<unknown> | P6NativeControl | undefined {
  return Array.from(children).filter(isInDefaultSlot).find(isControl) as P6Control<unknown> | P6NativeControl | undefined;
}
