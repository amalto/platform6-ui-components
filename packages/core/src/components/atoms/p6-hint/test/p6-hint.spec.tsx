import { newSpecPage } from '@stencil/core/testing';
import { enumArrayEntryToArray } from '../../../../shared/test/utils';
import { Mode, modes } from '../../../../shared/types';
import { P6Hint } from '../p6-hint';

describe('p6-hint', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [P6Hint],
      html: `<p6-hint>hint text</p6-hint>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  describe('mode', () => {
    it.each(enumArrayEntryToArray(modes))('has the class "is-%s" when mode is "%s"', async (name, value) => {
      const page = await newSpecPage({
        components: [P6Hint],
        html: `<p6-hint mode=${value}>${name}</p6-hint>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it('does not add class when mode is default', async () => {
      const page = await newSpecPage({
        components: [P6Hint],
        html: `<p6-hint mode=${Mode.default}>default</p6-hint>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it('does not add class when mode is undefined', async () => {
      const page = await newSpecPage({
        components: [P6Hint],
        html: `<p6-hint mode=${undefined}>undefined</p6-hint>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });
});
