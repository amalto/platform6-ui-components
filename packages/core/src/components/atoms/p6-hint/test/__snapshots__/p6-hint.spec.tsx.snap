// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-hint mode does not add class when mode is default 1`] = `
<p6-hint mode="2">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  default
</p6-hint>
`;

exports[`p6-hint mode does not add class when mode is undefined 1`] = `
<p6-hint mode="undefined">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  undefined
</p6-hint>
`;

exports[`p6-hint mode has the class "is-danger" when mode is "0" 1`] = `
<p6-hint class="is-danger" mode="0">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  danger
</p6-hint>
`;

exports[`p6-hint mode has the class "is-default" when mode is "2" 1`] = `
<p6-hint mode="2">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  default
</p6-hint>
`;

exports[`p6-hint mode has the class "is-info" when mode is "3" 1`] = `
<p6-hint class="is-info" mode="3">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  info
</p6-hint>
`;

exports[`p6-hint mode has the class "is-primary" when mode is "5" 1`] = `
<p6-hint class="is-primary" mode="5">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  primary
</p6-hint>
`;

exports[`p6-hint mode has the class "is-success" when mode is "4" 1`] = `
<p6-hint class="is-success" mode="4">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  success
</p6-hint>
`;

exports[`p6-hint mode has the class "is-warning" when mode is "1" 1`] = `
<p6-hint class="is-warning" mode="1">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  warning
</p6-hint>
`;

exports[`p6-hint renders 1`] = `
<p6-hint>
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  hint text
</p6-hint>
`;
